src/variety/pencils.js

```diff
@@ -582,314 +583,129 @@ FileIO:{
 // 正解判定処理実行部
 AnsCheck:{
 	checklist : [
-		"checkBranchLine",
-		"checkCrossLine",
-		"pencilTipLineEnd",
-		"crossPencilNumber",
-		"crossPencilBorder",
-		"pencilTipConnect",
-		"checkSmallArea",
-		
-		"pencilTwoEnds",
-		"pencilZeroLength",
-		"pencilSmallLength",
-		"pencilTipsOverlap",
-		"pencilNumberTooSmall",
-		
-		"pencilTipNoLine",
-		"lineNoTip",
-		"pencilLargeLength",
-		"unusedCells",
-		"pencilExactAreas",
-		"unusedBorders"
-	],
+		"checkTipHasPencil",
```

Comment:
Could you make sure these checks are at the top of the list?

* Lines cross or branch off
* A line crosses a pencil
* A line doesn't stop at a pencil tip (lcnt > 1)
* A line connects to two pencil tips
* A line is too long for this pencil
* A number is too large for this pencil

Reply:
I've implemented these changes in https://github.com/x-sheep/pzprjs/tree/pr-pencil. If it looks good to you, I'll add them to [my pending PR](https://github.com/robx/pzprjs/pull/40) and make it up-to-date with `master`.

Reply:
Yes please do! I'd be happy to get this merged soon. I should have time to merge this and make the pencil tip input change later this week. Sorry for the wait.